### PR TITLE
Improve Django settings and install notes

### PR DESCRIPTION
**Pull request by jraddaoui**
- Use the environment to configure the following Django settings:
  - ALLOWED_HOSTS
  - SECRET_KEY
  - DEBUG
- Fix static files configuration:
  - Remove static files from accesspoc.urls and trust Whitenoise.
  - Move existing static files inside accesspoc app.
  - Add STATIC_ROOT and change current STATICFILES_DIRS settings.
  - Add current general static folder to .gitignore.
  - Add `collectstatic` command notes to README.
- Serve uploaded ZIP files with Nginx:
  - Add secure view and URL to redirect to internal Nginx location using
    'X-Accel-Redirect'.
  - Add location to Nginx config in install notes and permissions notes.
  - Add 'FILE_UPLOAD_PERMISSIONS' Django setting.
  - Improve Docker Compose environment to reflect changes.
- Declare needed env. vars. in Tox and Docker Compose environments.
- Add technologies involved and sys. req. to README.
- Add install notes for Redis server and Celery worker.
- Add notes in README for new required env. vars. for production.

TODO:
- [x] Serve media files with Nginx in production.
- [x] Add installation notes for Redis and Celery.
- [x] Include system requirements in README.

Connects to #43.
Connects to #48.
Connects to #15.

### CODE REVIEW

**Comment by jraddaoui** [comment]
Hi @sevein, this is ready for another round of review. Sorry for the back and forward in some commits.

File: README.md [hunk 1]
```diff
@@ -56,9 +62,51 @@ By default, the application has five levels of permissions:
 
 For more information check the [user management and permissions feature file](features/user_management_and_permissions.feature).
 
+## Technologies involved
+
+The DIP Access Interface (real name pending) is a Django application that uses Elasticsearch 6.x as search engine, Celery 4.2 to process asynchronous tasks, a SQLite database and Redis as message broker (probably in the future, as cache system too).
+
+### Django, Celery and SQLite
+
+The Django application and the Celery worker need access to the source code, the database and the stored ZIP files. To avoid complexity and because the application currently uses SQLite as the database engine, it’s recommended to have both components working on the same server.
```
Comment (sevein) [inline README.md]: Perhaps "running on the same machine" or "sharing the same machine"?

File: README.md [hunk 2]
```diff
@@ -56,9 +62,51 @@ By default, the application has five levels of permissions:
...
+
+The application is ready to be served with Gunicorn and Gevent, using Whitenoise to serve the static files and Nginx, to proxy the application and serve the uploaded ZIP files. Check the install notes bellow for more information. Gunicorn is deployed using the Gevent worker class, meaning that a single worker should scale sufficiently even during I/O-bound operations like serving static files. If there are more CPU-bound tasks needed in the future, those will be delegated to the Celery async. task queue to ensure that the event loop is not blocked; therefore, the recommended amount of workers deployed is one.
```
Comment (sevein) [inline README.md]: s/bellow/below

File: README.md [hunk 3]
```diff
@@ -56,9 +62,51 @@ By default, the application has five levels of permissions:
...
+
+At this point, the application stores the uploaded ZIP files in the "media" folder at the application location. This should be considered to determine the disk capacity needed to hold the application data; in addition to the SQLite database, the space needed for the METS files extraction (mentioned above) and around 200 megabytes to hold the source code and Python dependencies.
+
+### Redis
+
+Redis is used as broker in the current Celery implementation and it will probably be used as cache system in the future. This component could be installed in the same or a different server and it’s URL can be configured through an environment variable read in the Django settings. At this point the memory footprint, the CPU usage and the disk allocation needed for snapshots will be minimal. Check [the Redis FAQ page](https://redis.io/topics/faq) for more information.
```
Comment (sevein) [inline README.md]: s/it's/its

File: README.md [hunk 4]
```diff
@@ -56,9 +62,51 @@ By default, the application has five levels of permissions:
...
+
+### Elasticsearch
+
+Elasticsearch could also be installed in the same or different servers and it’s URL(s) can be configured through an environment variable read in the Django settings. The application expects Elasticsearch 6.x, which requires at least Java 8 in order to run. Only Oracle’s Java and the OpenJDK are supported and the same JVM version should be used on all Elasticsearch nodes and clients.
```
Comment (sevein) [inline README.md]: s/it's/its

File: README.md [hunk 5]
```diff
@@ -56,9 +62,51 @@ By default, the application has five levels of permissions:
...
+
+The Elasticsearch node/cluster configuration can be fully customized, however, for the current implementation, a single node with the the default JVM heap size of 1GB set by Elasticsearch would be more than enough. It could even be reduced to 512MB if more memory is needed for other parts of the application or to reduce its requirements. For more info on how to change the Elasticsearch configuration check [their documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html), specially [the JVM heap size page](https://www.elastic.co/guide/en/elasticsearch/reference/current/heap-size.html).
+
+The Elasticsearch indexes size will vary based on the application data and they will require some disk space, but it’s hard to tell how much at this point.
```
Comment (sevein) [inline README.md]: Should we explain how to adjust the number of shards and replicas of the indices so their status is not "yellow"? I'm not sure if that's still a thing in ES 6.x but I remember that was needed in older versions of ES.

File: README.md [hunk 6]
```diff
@@ -165,6 +216,13 @@ export $(cat ~/accesspoc-env)
 cd accesspoc
 ```
 
+Create the 'media' folder with read and execute permissions for the group:
```
Comment (sevein) [inline README.md]: Use backticks is preferred, `media` instead of 'media'.

File: README.md [hunk 7]
```diff
@@ -191,15 +249,69 @@ Compile translation files:
...
+            --pidfile=${CELERYD_PID_FILE} \
+            --logfile=${CELERYD_LOG_FILE} \
+            --loglevel=WARNING'
+ExecReload=/bin/sh -c '${CELERY_BIN} multi restart accesspoc-worker \
+            -A accesspoc \
+            --pidfile=${CELERYD_PID_FILE} \
+            --logfile=${CELERYD_LOG_FILE} \
+            --loglevel=WARNING'
+ExecStop=/bin/sh -c '${CELERY_BIN} multi stopwait accesspoc-worker \
+            --pidfile=${CELERYD_PID_FILE}'
```
Comment (sevein) [inline README.md]: Is `sh` needed?

File: README.md (hunk 2 above)
Comment (sevein) [reply README.md]: s/Whitenoise/WhiteNoise

File: etc/nginx/nginx.conf [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      location / {
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_redirect off;
+        proxy_buffering off;
+        proxy_pass http://accesspoc:8000;
+      }
+    }
+
+}
```
Comment (sevein) [inline etc/nginx/nginx.conf]: Nitpick: inconsistent indentation 2 vs 4 whitespaces.

File: README.md (hunk 7 above)
Comment (jraddaoui) [reply README.md]: I guess not, will check.

Comment (jraddaoui) [reply README.md]: It's needed when using an env. var. to declare the path, as `${...` is not recognized as an absolute path, I'll not use the env. var. for the executable.

File: README.md (hunk 5 above)
Comment (jraddaoui) [reply README.md]: Not an issue anymore, not sure if a side effect of having three indexes ...

```
curl -XGET http://localhost:9200/_cluster/health?pretty=true
{
  "cluster_name" : "elasticsearch",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 3,
  "active_shards" : 3,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 100.0
}
```

**Pull request merged**